internal/terraform/terraform.go

```diff
@@ -73,21 +74,63 @@ func GetResourceType(resource hclwrite.Block) string {
 	return resource.Labels()[0]
 }
 
-func ValidateTerraformInitRun(dir string) error {
-	_, err := os.Stat(dir + "/.terraform")
+func ValidateInitRun(dir string, terragrunt bool) error {
+	path := dir + "/."
+	if terragrunt {
+		path += "terragrunt-cache"
+	} else {
+		path += "terraform"
+	}
+
+	if _, err := os.Stat(path); err != nil {
+		var errorTypeStr string
+		if terragrunt {
+			errorTypeStr = "terragrunt"
+		} else {
+			errorTypeStr = "terraform"
+		}
 
-	if err != nil {
 		if os.IsNotExist(err) {
-			return errors.New("terraform init must run before running terratag")
+			return fmt.Errorf("%s init must run before running terratag", errorTypeStr)
 		}
 
-		return fmt.Errorf("couldn't determine if terraform init has run: %v", err)
+		return fmt.Errorf("couldn't determine if %s init has run: %v", errorTypeStr, err)
 	}
 
 	return nil
 }
 
-func GetTerraformFilePaths(rootDir string) ([]string, error) {
+func GetFilePaths(dir string, terragrunt bool) ([]string, error) {
+	if terragrunt {
+		return getTerragruntFilePath(dir)
+	} else {
+		return getTerraformFilePaths(dir)
+	}
+}
+
+func getTerragruntFilePath(rootDir string) ([]string, error) {
+	rootDir += "/.terragrunt-cache"
+
+	var tfFiles []string
+	if err := filepath.WalkDir(rootDir, func(path string, d fs.DirEntry, err error) error {
```

Comment:
Consider avoid using inline syntax for this function declaration. I find it a little confusing, especially for newcomers to golang

Reply:
It's actually done in this way on purpose (all examples are like this) and is standard in go. The main reason is that I need to use the variables in lines 112 and 114.
I could move it to a function that returns a function (to store the variable's context) if you think it will make it more readable. Let me know what you prefer.

Reply:
This is no biggie so if you feel it's a valid standard then let's leave it be 👍